mewarpx/medocker/docker_shared/run_warpx.sh

```diff
@@ -119,21 +156,31 @@ do
     find . -type f -exec touch -a -c {} +
 done
 }
+
 # https://stackoverflow.com/questions/21465297/tee-stdout-and-stderr-to-separate-files-while-retaining-them-on-their-respective
-{ { bash $SCRIPTNAME; } > >( tee -a stdout.out ); } \
+bash $SCRIPTNAME > >( tee -a stdout.out ) \
                        2> >( tee -a stderr.err >&2 ) &
 
+# sleep so that there is time for the mpiexec process to launch
+sleep 2
+
 MAINID=$!
+MPIEXECID=$(pgrep -P $MAINID -f mpiexec)
+echo "main process id is $MAINID"
+echo "mpiexec process id: $MPIEXECID"
 checkpoint &
 CHECKID=$!
+pingterminatewarning &
+PINGID=$!
 wait -n $MAINID $CHECKID
 EXITFLAG=$?
 # kill, but silently, as one will error no matter what. It looks like killing
 # $MAINID actually isn't enough -- child processes still run -- but it
 # shouldn't matter as the docker container should exit when this script exits,
 # I think.
```

Comment:
```suggestion
# kill, but silently, as either CHECKID or MPIEXECID kill will error no matter
# what. Note MPIEXECID is klled with -KILL to prevent WarpX from trying to
# write a final checkpoint, which would fail.
```